src/PrestaShopBundle/Resources/config/routing/admin/configure/advanced_parameters/email.yml

```diff
@@ -14,6 +14,14 @@ admin_emails_search:
         _legacy_controller: AdminEmails
         _legacy_link: AdminEmails:search
 
+admin_emails_save_options_get:
+    path: /options
+    methods: [GET]
+    defaults:
+        _controller: 'PrestaShopBundle:Admin\Configure\AdvancedParameters\Email:index'
+        _legacy_controller: AdminEmails
+        _legacy_link: AdminEmails:update
+
 admin_emails_save_options:
```

Comment:
I changed all option forms when simplifying so instead of always redirecting back to index after saving they render from instead. This allowed usage of constraints in forms, but now as found by Q&A, it introduces one problem. 

The problem happens when you try to save the form, you get redirected to sub url, like /options here and stay there. If you do some sort of redirection to the same page(Changing pagination if there is a list, changing shop if multishop etc.) you get redirected to GET of this url which it doesn't handle and you get an exception. 

So here is the simplest solution, add alternate GET routes to all option saving routes which all would be use an index action. With this if you save options and keep doing things on the page you will stay in /options url, I am not sure about the exact implications of that but I guess if there is another post action that redirects back to the same page it could cause problems.

I see another 2 solutions.
1) Go back to always redirecting back to index after saving the form. Which means constraints once again won't work, but maybe it's better than messing with routes here.
2) Figure out a way to change actions like changing shop in multishop to redirect you back to the index of the options page you are at, instead of the exact url you are at. This sounds like it could be complicated and introduce more problems, so unless this is needed for some other reason I am not sure if it's worth pursuing. 

So I would love to get more oppinions on this.

Reply:
So my memory fails me 😅  did we say we aim for "no constraint error messages" in the end?

Reply:
Yeah we decided to go back to not using constraint messages.

https://github.com/PrestaShop/ADR/pull/17/files

Reply:
Actually, the ADR is not approved. So it's not a decision.

Reply:
@Progi1984 This ADR was not meant to be a decision, rather a simple record.

Initially we were not using constraint messages. Then we tried to start to use them. Now we see a huge limitation thanks to constraint messages so we go back to initial structure.

The ADR PR was only meant to store a record of this attempt to document it, so that next generation of PrestaShop developers will not do the same issue as we did. No vote needed, there is nothing to decide on.